src/cpp/CMakeLists.txt

```diff
@@ -346,9 +346,11 @@ if(TINYXML2_SOURCE_DIR)
 endif()
 
 if(ANDROID)
-    list(APPEND ${PROJECT_NAME}_source_files
-        ${ANDROID_IFADDRS_SOURCE_DIR}/ifaddrs.c
-        )
+    if((ANDROID_PLATFORM LESS_EQUAL 23) OR (ANDROID_NATIVE_API_LEVEL LESS_EQUAL 23))
```

Comment:
I think we need a similar protection on the root [CMakeLists.txt](https://github.com/eProsima/Fast-DDS/blob/c8a9f196517d463d4cbfe9085247ad2c9aa00b54/CMakeLists.txt#L170-L172)

Reply:
Added the same check to the root file as requested.